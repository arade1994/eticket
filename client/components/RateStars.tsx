interface Props {
  rate: number;
}

const RateStars = ({ rate }: Props) => {
  return Array.from(Array(5).keys()).map((index) => (
    <div id="starsContainer" className="rating" style={{ marginBottom: 0 }}>
      <label
        aria-label="star"
        htmlFor={`star${index}`}
        className={`star ${index + 1 <= rate ? "checked" : ""}`}
      >
        &#9733;
      </label>
    </div>
  ));
};

export default RateStars;
