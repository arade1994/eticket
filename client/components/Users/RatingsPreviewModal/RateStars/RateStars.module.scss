.starsContainer {
  display: flex;
  margin-left: 7px;

  & .rate {
    margin-bottom: 0px;
    display: flex;
    flex-direction: row-reverse;
    justify-content: center;

    & input {
      display: none;
    }

    & .star {
      cursor: pointer;
    }

    & .checked {
      color: orange;
    }
  }
}
