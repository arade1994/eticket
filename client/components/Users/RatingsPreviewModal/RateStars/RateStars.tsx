import cx from "classnames";
import classes from "./RateStars.module.scss";

interface Props {
  rate: number;
}

const RateStars = ({ rate }: Props) => {
  return (
    <div className={classes.starsContainer}>
      {Array.from(Array(rate).keys()).map((index) => (
        <div key={index} className={classes.rate}>
          <label
            aria-label="star"
            htmlFor={`star${index}`}
            // className={`star ${index + 1 <= rate ? "checked" : ""}`}
            className={cx(
              classes.star,
              index + 1 <= rate ? classes.checked : ""
            )}
          >
            &#9733;
          </label>
        </div>
      ))}
    </div>
  );
};

export default RateStars;
