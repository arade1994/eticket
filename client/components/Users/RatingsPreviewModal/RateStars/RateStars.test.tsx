import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import RateStars from "./Users/RatingsPreviewModal/RateStars/RateStars";

describe("<RateStars />", () => {
  test("It should display rate stars component as expected", () => {
    const { baseElement } = render(<RateStars rate={4} />);

    expect(baseElement).toMatchSnapshot();

    const stars = screen.getAllByLabelText("star");
    expect(stars.length).toEqual(5);
    const selectedStars = stars.filter((star) =>
      star.classList.contains("checked")
    );
    expect(selectedStars.length).toEqual(4);
  });
});
