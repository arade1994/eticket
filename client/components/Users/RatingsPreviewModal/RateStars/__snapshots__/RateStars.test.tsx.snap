// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<RateStars /> > It should display rate stars component as expected 1`] = `
<body>
  <div>
    <div
      class="_starsContainer_21043f"
    >
      <div
        class="_rate_21043f"
      >
        <label
          aria-label="star"
          class="_star_21043f _checked_21043f"
          for="star0"
        >
          ★
        </label>
      </div>
      <div
        class="_rate_21043f"
      >
        <label
          aria-label="star"
          class="_star_21043f _checked_21043f"
          for="star1"
        >
          ★
        </label>
      </div>
      <div
        class="_rate_21043f"
      >
        <label
          aria-label="star"
          class="_star_21043f _checked_21043f"
          for="star2"
        >
          ★
        </label>
      </div>
      <div
        class="_rate_21043f"
      >
        <label
          aria-label="star"
          class="_star_21043f _checked_21043f"
          for="star3"
        >
          ★
        </label>
      </div>
    </div>
  </div>
</body>
`;
